runtime/compiler/optimizer/J9EstimateCodeSize.cpp

```diff
@@ -1730,10 +1734,40 @@ TR_J9EstimateCodeSize::realEstimateCodeSize(TR_CallTarget *calltarget, TR_CallSt
                   }
                else
                   {
+                  // Now check the invocation count for the callee if interpreted.
+                  bool isInterpretedCallWithLowFrequency = false;
+                  static bool shouldNotInlineInterpretedMethods = feGetEnv("TR_DisableSkippingInliningOfColdInterpretedMethods") == NULL;
+                  if (shouldNotInlineInterpretedMethods)
+                     {
+                     static bool includeWarmCompilationsAsWell = feGetEnv("TR_EnableSkippingInliningOfColdInterpretedMethodsInWarm") != NULL;
+                     if (targetCallee->_calleeMethod->isInterpretedForHeuristics() && !comp()->fej9()->compiledAsDLTBefore(targetCallee->_calleeMethod))
+                        {
+                        if (comp()->getMethodHotness() > warm)
+                           {
+                           static const char *cutOffHotAndAbove = feGetEnv("TR_FreqCutOffForInterpretedCalleeInHotAndAbove");
+                           static const int32_t cutOffFreqForHotAndAbove = cutOffHotAndAbove ? atoi(cutOffHotAndAbove) : FREQ_CUTOFF_INTERPRETED_HOTANDABOVE;
+                           isInterpretedCallWithLowFrequency = currentBlock->getFrequency() < cutOffFreqForHotAndAbove;
+                           }
+                        else if (includeWarmCompilationsAsWell && comp()->getMethodHotness() == warm)
+                           {
+                           static const char *cutOffForWarm = feGetEnv("TR_FreqCutOffForInterpretedCalleeInWarm");
+                           static const int32_t cutOffFreqForWarm = cutOffForWarm ? atoi(cutOffForWarm) : FREQ_CUTOFF_INTERPRETED_WARM;
+                           isInterpretedCallWithLowFrequency = profileManager->getCallGraphProfilingCount(targetCallee->_calleeMethod->getPersistentIdentifier(),
```

Comment:
Why are we using the `getCallGraphProfilingCount` api here whereas we use the block frequency on the other side of this conditional, i.e. for hot and above ?

The reason I ask is that @mpirvu and I are considering if we should remove the `getCallGraphProfilingCount` currently via experiments.

Reply:
> With removal of getCallGraphProfilingCount - for warm / cold compilations - what interface we are looking to utilize

Why isn't block frequency enough to determine the call count?

Reply:
That is the cleanest position to take and @mpirvu and trying to take that position for now at least, as we try to remove consumers of these "call graph profiling count" apis. For example, we might be interested in removing the `isColdCall` api Rahil mentioned and replace it with whatever amended block frequency based test is suitable.

Reply:
Thanks @mpirvu  - I do agree and as I mentioned, when making change, and inspecting what we get from block frequency and call graph count is more or less in same range, and as @vijaysun-omr  mentioned, there is a plan to remove the `getCallGraphProfilingCount`, we can simplify and use the block frequency for warm as well. Making the change.

Reply:
I have made change to remove using getProfilingCallCount and use block frequency for the interpreted targets in https://github.com/eclipse-openj9/openj9/compare/58919c22da5348931f21048a9e30c0075d8d5efa..d386800e13f70f41f0ad960a8230a6bed31aa5a6

To ensure, I do not miss, interest in removing `isColdCall` API usage is a separate set of changes we would be making - not part of this PR, right?

Reply:
> interest in removing isColdCall API usage is a separate set of changes we would be making - not part of this PR, right?

Correct